### ISSUE TITLE
Cannot find module 'publish'

### ISSUE THREAD

**Issue by nightpool** [englercj/tsd-jsdoc#18]
I'm trying to run jsdoc against an already existing library, but I get the error `FATAL: Unable to load template: Cannot find module './node_modules/tsd-jsdoc/publish'`. I've confirmed that npm install worked, but it looks like the publish file was removed in this commit: https://github.com/englercj/tsd-jsdoc/commit/c8b7f3c99b39ab90568c2116b24497ffe76146a8

this is the command I'm using: `jsdoc -c jsdoc.json -t node_modules/tsd-jsdoc`

How can I get tsd-jsdoc running now? Do I have to compile it manually, or use a different source?

**Comment by nightpool** [comment]
Using `-t node_modules/tsd-jsdoc/lib` instead leads to this error:

```
Generating output files...
node_modules/dts-dom/bin/index.js:385
            switch (e.kind) {
                     ^

TypeError: Cannot read property 'kind' of null
    at writeReference (node_modules/dts-dom/bin/index.js:385:22)
    at writeParameter (node_modules/dts-dom/bin/index.js:485:9)
    at writeDelimited (node_modules/dts-dom/bin/index.js:494:13)
    at writeFunction (node_modules/dts-dom/bin/index.js:473:9)
    at writeDeclaration (node_modules/dts-dom/bin/index.js:650:28)
    at Object.emit (node_modules/dts-dom/bin/index.js:229:5)
    at Emitter.emit (node_modules/tsd-jsdoc/lib/Emitter.js:65:24)
    at Object.publish (node_modules/tsd-jsdoc/lib/publish.js:25:39)
    at Object.module.exports.cli.generateDocs (node_modules/jsdoc/cli.js:430:39)
    at Object.module.exports.cli.processParseResults (node_modules/jsdoc/cli.js:383:20)
    at module.exports.cli.main (node_modules/jsdoc/cli.js:227:14)
    at Object.module.exports.cli.runCommand (node_modules/jsdoc/cli.js:180:5)
    at node_modules/jsdoc/jsdoc.js:103:9
    at Object.<anonymous> (node_modules/jsdoc/jsdoc.js:104:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
```

**Comment by nightpool** [comment]
i should note that I'm using the latest `2.0.0-beta.1` from npm

**Comment by englercj** [comment]
Hmm, looks like jsdoc expects the publish.js file to be in the root folder of the template directory. I assumed it used the `main` property of the package.json but I guess not.

As for the e.kind is null error, can you give me a js block that reproduces it?

**Comment by nightpool** [comment]
For the first, I get a very similar error: `FATAL: Unable to load template: Cannot find module 'tsd-jsdoc/publish`

For the second, here's a minimal reproduction:

```js
/**
 * Duplicates the passed string
 * @param  {string} foo - the string to duplicate
 * @return {string} the duplicated string
 */
function dupeStr(foo) {
    return foo + foo;
}
```

**Comment by nightpool** [comment]
also I can't get the tests to work at all.....

**Comment by englercj** [comment]
Thanks for trying that, I'll move the publish script around.

There aren't any "tests" yet, all `npm test` does is run on the js files in there and output a file that I can look at manually. No automated testing yet. When you say "I can't get the tests to work all all" can you tell me what that means? Are you running on a cloned repo or the installed package? The installed package doesn't include the tests or anything.

I am able to reproduce the error you are getting though when used with the example you provided and no other code. Looking into that now.

**Comment by nightpool** [comment]
I mean that if I clone the repo, do npm install, npm build, and npm test, I get an error.

**Comment by nightpool** [comment]
Also, to be clear, I reduced down the minimal test case I provided earlier from ~1,000 lines of module code. The problem is clearly not that it's standalone.

**Comment by englercj** [comment]
> The problem is clearly not that it's standalone.

When I ran it with all my other code it worked, I only got the error when I ran it on only this code piece. Turns out it was due to a logic error in how I handle the "private" config which is probably different in your config than in the repo's.

> I get an error.

Can you share what the error is?

---

This should be fixed now in `v2.0.0-beta.2`.

**Comment by nightpool** [comment]
@englercj the error is in the comment

**Comment by englercj** [comment]
So the original error you got running on your code, was also the same error you got when running the tests?

**Comment by nightpool** [comment]
No, that's a different error.

**Comment by nightpool** [comment]
![image](https://cloud.githubusercontent.com/assets/233815/23047480/187577fe-f47e-11e6-846d-41218d1b90e3.png)

**Comment by englercj** [comment]
Looks like dts-dom hasn't released my additions that add support for indexSignature. I'm locally using `HEAD` and not the tagged release, which is why it works for me.

Hopefully they release it soon.

**Comment by englercj** [comment]
Ok, released, as long as you have `0.0.15` installed it should be good to go. Let me know if you have any more issues, open as many as you want. I appreciate you trying out the beta.

**Comment by Nols1000** [comment]
@englercj This error occures in version `2.0.0-beta.5` again.
